srv/salt/_modules/rgw.py

```diff
@@ -262,6 +262,8 @@ def s3connect(user):
     """
     Return an S3 connection
     """
+    if access_key(user) is None or secret_key(user) is None:
```

Comment:
There is a specific case where this might be not true, if LDAP is used for RGW authentication; then secret_key might be empty. Does boto module throw the same exception for empty secret keys?

Reply:
I tried with empty access_key and secret_key. Surprisingly, boto did not complain in that case. It seems like the issue we saw was because we returned None here (https://github.com/SUSE/DeepSea/blob/master/srv/salt/_modules/rgw.py#L157) If I change the return just to empty string it works. Boto does not complain with empty keys.

Reply:
I think this PR will work for LDAP where we have empty keys. In that case, https://github.com/SUSE/DeepSea/blob/master/srv/salt/_modules/rgw.py#L155 will return empty string, and boto s3connect will not complain. create_bucket/lookup will handle the exception. In case there is no user json file, then it make sense to not call s3connect at all, and return.